tests/bdp_spark_pi_submit_test.erl

```diff
@@ -0,0 +1,155 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% @doc A module to test Spark fail/recovery under BDP service manager.
+
+-module(bdp_spark_pi_submit_test).
+-behavior(riak_test).
+-export([confirm/0]).
+-include_lib("eunit/include/eunit.hrl").
+
+%spark master definitions
+-define(SPARK_MASTER_SERVICE_NAME, "my-spark-master").
+-define(SPARK_MASTER_SERVICE_TYPE, "spark-master").
+-define(SPARK_MASTER_SERVICE_CONFIG, [{"HOST", "127.0.0.1"},{"RIAK_HOST","127.0.0.1:8087"}]).
+%spark worker definitions
+-define(SPARK_WORKER_SERVICE_NAME, "my-spark-worker").
+-define(SPARK_WORKER_SERVICE_TYPE, "spark-worker").
+-define(SPARK_WORKER_SERVICE_CONFIG, [{"MASTER_URL", "spark://127.0.0.1:7077"},{"SPARK_WORKER_PORT","8081"}]).
+
+
+% this code tests if a spark job can successfully be submitted to a two
+% node spark cluster within a three node bdp cluster
+
+% WARNINGS: spark worker configs must contain exact ip address of spark
+% master, not 127.0.0.1.  Otherwise, spark master and worker will fail
+% to link together.  As such, os:cmd is used to get the ip address of
```

Comment:
The comment re: method of getting the spark master's ip is out of date. The code uses inet:getifaddrs .